src/main/routes/receiver.ts

```diff
@@ -132,13 +132,16 @@ export default express.Router()
             await ClaimStoreClient.linkDefendant(user)
             res.redirect(await retrieveRedirectForLandingPage(user))
           } else {
-            Promise.all(user.getLetterHolderIdList().map(
-            (letterHolderId) => linkDefendantWithClaimByLetterHolderId(letterHolderId, user)
-            )
-          )
-            .then(async () => res.redirect(await retrieveRedirectForLandingPage(user)))
```

Comment:
The previous implementation resulted in retrying `retrieveRedirectForLandingPage` logic when an error was thrown in `retrieveRedirectForLandingPage` function. That behaviour is unexpected especially that we have retry on the HTTP library right now.